readthedocs/core/signals.py

```diff
@@ -62,4 +67,23 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
 
     return False
 
+
+@receiver(pre_delete, sender=User)
+def delete_projects_and_organizations(instance, *args, **kwargs):
+    # Here we count the owner list from the projects that the user own
+    # Then exclude the projects where there are more than one owner
+    projects_id = (instance.projects.all().annotate(num_users=Count('users'))
+                                          .exclude(num_users__gt=1)
+                                          .values_list('id', flat=True))
+
+    # Here we count the users list from the organization that the user belong
+    # Then exclude the organizations where there are more than one user
+    oauth_organizations_id = (instance.oauth_organizations.annotate(num_users=Count('users'))
+                                                          .exclude(num_users__gt=1)
+                                                          .values_list('id', flat=True))
+
+    bulk_delete_projects.delay(projects_id)
+    bulk_delete_oauth_remote_organizations.delay(oauth_organizations_id)
```

Comment:
If there is no strong reason that either of these need to be executed via a task, we can just remove them directly here.